stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,59 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), copy(B); sortby)
+end
+function eigen!(A::AbstractMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    D = copy(B.D)
+    M = B.uplo == 'U' ? B.U : B.L
+    A .= A[B.p, B.p]
+    ldiv!(M, A)
+    rdiv!(A, M')
+    ldiv!(lu!(D), A)
+    vals, vecs = eigen!(A; sortby)
+    vecs .= ldiv!(M', vecs)[invperm(B.p), :]
+    GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
+end
+function eigen!(A::StridedMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
```

Comment:
Is this method really needed? The `[l/r]div!` etc. above should use those LAPACK functions via multiple dispatch.

Reply:
Thank you for this comment. Please see my post below for my response.